envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ServiceTagsAndCanaryTest.kt

```diff
@@ -109,19 +123,29 @@ open class ServiceTagsAndCanaryTest : EnvoyControlTestConfiguration() {
         assertThat(stats.ipsumRegularHits).isEqualTo(0)
     }
 
-    protected open fun callStats() = CallStats(listOf(loremCanaryContainer, loremRegularContainer, ipsumRegularContainer))
+    fun waitForReadyServices(vararg serviceNames: String) {
+        serviceNames.forEach {
+            untilAsserted {
+                envoy.egressOperations.callService(it).also {
+                    assertThat(it).isOk()
+                }
+            }
+        }
+    }
 
-    protected open fun callEchoServiceRepeatedly(
+    fun callEchoServiceRepeatedly(
```

Comment:
Shouldn't those methods be private?

Reply:
I like the Spock approach of keeping the test code free of all the clutter, including unnecessary keywords. Since nobody will reference test classes it's fine to keep the default visibility, with the benefit of fewer words to interpret on the screen. We can discuss if everyone agrees with this approach.